pkg/processor/schemaregistry/client_fake_test.go

```diff
@@ -0,0 +1,344 @@
+	fr.init()
+	fr.m.Lock()
+	defer fr.m.Unlock()
+
+	return fr.findBySubjectVersion(subject, version)
+}
+
+func (fr *fakeRegistry) SubjectVersionsByID(id int) []sr.SubjectSchema {
+	fr.init()
+	fr.m.Lock()
+	defer fr.m.Unlock()
+
+	return fr.findAllByID(id)
+}
+
+func (fr *fakeRegistry) nextID() int {
+	fr.idSequence++
+	return fr.idSequence
+}
+
+func (fr *fakeRegistry) nextVersion(subject string) int {
+	return len(fr.findBySubject(subject)) + 1
+}
+
+func (fr *fakeRegistry) findBySubject(subject string) []sr.SubjectSchema {
+	var sss []sr.SubjectSchema
+	for _, ss := range fr.schemas {
+		if ss.Subject == subject {
+			sss = append(sss, ss)
+		}
+	}
+	return sss
+}
+
+func (fr *fakeRegistry) findOneByID(id int) (sr.Schema, bool) {
+	for _, ss := range fr.schemas {
+		if ss.ID == id {
+			return ss.Schema, true
+		}
+	}
+	return sr.Schema{}, false
+}
+
+func (fr *fakeRegistry) findAllByID(id int) []sr.SubjectSchema {
+	var sss []sr.SubjectSchema
+	for _, ss := range fr.schemas {
+		if ss.ID == id {
+			sss = append(sss, ss)
+		}
+	}
+	return sss
+}
+
+func (fr *fakeRegistry) findBySubjectID(subject string, id int) (sr.SubjectSchema, bool) {
+	for _, ss := range fr.schemas {
+		if ss.Subject == subject && ss.ID == id {
+			return ss, true
+		}
+	}
+	return sr.SubjectSchema{}, false
+}
+
+func (fr *fakeRegistry) findBySubjectVersion(subject string, version int) (sr.SubjectSchema, bool) {
+	for _, ss := range fr.schemas {
+		if ss.Subject == subject && ss.Version == version {
+			return ss, true
+		}
+	}
+	return sr.SubjectSchema{}, false
+}
+
+// fakeServer is a fake schema registry server.
+type fakeServer struct {
+	mux  http.ServeMux
+	fr   fakeRegistry
+	logf func(format string, args ...any)
+}
+
+func newFakeServer(logf func(format string, args ...any)) *fakeServer {
+	fs := &fakeServer{
+		logf: func(format string, args ...any) { /* no op */ },
+	}
+	if logf != nil {
+		fs.logf = logf
+	}
+	fs.mux.Handle("/schemas/ids/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tokens := strings.Split(r.URL.EscapedPath(), "/")
+		switch {
+		case len(tokens) == 4:
+			fs.schemaByID(w, r)
+		case len(tokens) == 5 && tokens[4] == "versions":
+			fs.subjectVersionsByID(w, r)
+		default:
+			http.NotFound(w, r)
+		}
+	}))
+	fs.mux.Handle("/subjects/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tokens := strings.Split(r.URL.EscapedPath(), "/")
+		switch {
+		case len(tokens) == 4 && tokens[3] == "versions":
+			fs.createSchema(w, r)
+		case len(tokens) == 5 && tokens[3] == "versions":
+			fs.schemaBySubjectVersion(w, r)
+		default:
+			http.NotFound(w, r)
+		}
+	}))
+	return fs
+}
+
+func (fs *fakeServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	fs.logf("%s %s", r.Method, r.RequestURI)
+	fs.mux.ServeHTTP(w, r)
+}
+
+func (fs *fakeServer) createSchema(w http.ResponseWriter, r *http.Request) {
+	// POST /subjects/{subject}/versions => returns ID
+	if r.Method != http.MethodPost {
+		http.NotFound(w, r)
```

Comment:
You're right, it should be 405. I think I used 404 because by default any unrecognized path/method combination ends up returning 404 (I even simplified this in a commit down the line, see https://github.com/ConduitIO/conduit/pull/1096/commits/9e0fcbd9754e2295ffc4583d41ae505b2bc10e62#diff-b9f542e140b2f8ad7e634f3244865bd94f526c8b43d7d6b9eb3bd4b405401991R216). In the end, what matters is that a test fails if it tries to reach an endpoint that does not exist, this should still be the case, as the body will be empty 🙂